barretenberg/cpp/src/barretenberg/execution_trace/execution_trace.cpp

```diff
@@ -56,10 +74,13 @@ typename ExecutionTrace_<Flavor>::TraceData ExecutionTrace_<Flavor>::construct_t
     Builder& builder, typename Flavor::ProvingKey& proving_key, bool is_structured)
 {
     ZoneScopedN("construct_trace_data");
-    TraceData trace_data{ builder, proving_key };
 
-    // Complete the public inputs execution trace block from builder.public_inputs
-    populate_public_inputs_block(builder);
+    if constexpr (IsPlonkFlavor<Flavor>) {
```

Comment:
only for plonk since for honk, we do in DeciderProvingKey(). We want to do it there for Honk because we want the trace to be completed before computing offsets.